scripts/create_istio_system.sh

```diff
@@ -1,37 +1,49 @@
 #!/bin/bash
-istioctl version
-old_versions=$(kubectl get --ignore-not-found=true deploy istio-pilot -n istio-system -o=jsonpath='{$.spec.template.spec.containers[*].image}')
-# check if older version of istio is still installed in istio-system namespace
-if [[ $old_versions == *"1.3"* || $old_versions == *"1.4"* || $old_versions == *"1.5"* ]]; then
-    echo "Please uninstall current istio version before attempting to install $ISTIO_VERSION"
+istioctl version -i $ISTIO_NAMESPACE
+if [[ $FORCE_INSTALL == false ]]; then
+  old_versions=$(kubectl get --ignore-not-found=true deploy istio-pilot -n istio-system -o=jsonpath='{$.spec.template.spec.containers[*].image}')
+  # check if older version of istio is still installed in istio-system namespace
+  if [[ $old_versions == *"1.3"* || $old_versions == *"1.4"* || $old_versions == *"1.5"* ]]; then
+      echo "please uninstall current istio version before attempting to install $ISTIO_VERSION"
+      exit 1
+  fi
+  # force install will skip file checks. same analyze command is valid for install and upgrade
+  if test -f "/app/overlay-config.yaml"; then
+    echo "running istioctl analyze with overlay"
+    istioctl analyze /app/istio-base.yaml /app/overlay-config.yaml -i $ISTIO_NAMESPACE --failure-threshold Error
+  else
+    echo "running istioctl analyze"
+    istioctl analyze /app/istio-base.yaml -i $ISTIO_NAMESPACE --failure-threshold Error
+  fi
+  if [[ $? -ne 0 ]]; then
+    echo "error found during istioctl analyze"
+    exit 1
+  fi
 fi
 
-versions=$(kubectl get --ignore-not-found=true deploy istiod -n istio-system -o=jsonpath='{$.spec.template.spec.containers[*].image}')
-if [[ $CANARY_REVISION ]] || [[ $versions == "" ]]
-then
-    if test -f "/app/overlay-config.yaml"
-    then
-        echo "Running istioctl install with overlay"
+versions=$(kubectl get --ignore-not-found=true deploy istiod -n $ISTIO_NAMESPACE -o=jsonpath='{$.spec.template.spec.containers[*].image}')
+if [[ $CANARY_REVISION ]] || [[ $versions == "" ]]; then
+    if test -f "/app/overlay-config.yaml"; then
+        echo "running istioctl install with overlay"
```

Comment:
Do we need several log statements for `with overlay`? If that's not required, I think that it would be better to use a bash array to store these values instead.

At the top of the file:
```
ISTIO_FILES=("/app/istio-base.yaml")
if test -f "/app/overlay-config.yaml"; then
    echo "creating istio system using overlay file at /app/overlay-config.yaml"
    ISTIO_FILES+=("/app/overlay-config.yaml")
fi
```

And then instead of multiple if-else statements, we can just do something like:
```
istioctl analyze $ISTIO_FILES -i $ISTIO_NAMESPACE --failure-threshold Error
```
or
```
istioctl install -y ${ISTIO_FILES[@]/#/-f }
```